packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -107,9 +112,11 @@ class SimControl {
     final Map<String, dynamic> devicesSection = await _list(SimControlListSection.devices);
 
     for (String deviceCategory in devicesSection.keys) {
-      final List<dynamic> devicesData = devicesSection[deviceCategory] as List<dynamic>;
-      for (Map<String, dynamic> data in devicesData.map<Map<String, dynamic>>(castStringKeyedMap)) {
-        devices.add(SimDevice(deviceCategory, data));
+      final Object devicesData = devicesSection[deviceCategory];
+      if (devicesData != null && devicesData is List<dynamic>) {
```

Comment:
nit: Flip the sense of the if and unindent the nested for loop:
```dart
if (devicesData == null || devicesData is! List<dynamic>) {
  continue;
}
...
```

Reply:
discussed offline: dart will not type promote outside of the block to `List<dynamic>`, thus requiring another cast